### ISSUE TITLE
C#: <see cref="X" /> on type should prefer to look for its members

### ISSUE THREAD

**Issue by exyi** [doxygen/doxygen#7447]
I have this class in my project:

```C#

/// <summary> <see cref="Name">, <see cref="Accessibility" /></summary>
public partial class TypeSignature
{
    ...
   public string Name { get; }
   public Accessibility Accessibility { get; }
 }  

```

Doxygen generates description "Name, Accessibility", where Name is link to the property `Name`, but `Accessibility` is a link to the class `Accessibility`. C# compiler however generates link to the property `Accessibility`:

```XML
<summary> <see cref="P:Coberec.ExprCS.TypeSignature.Name" />, <see cref="P:Coberec.ExprCS.TypeSignature.Accessibility" /> </summary>
```

**Comment by albert-github** [comment]
With version 1.8.16 I see in the brief description a link with `Name` and and with second `Accessibility` in the brief line `Accessibility Accessibility`. Do I overlook something? Maybe an image or the resulting output (with indication where the link is missing) would help to clarify the issue.
- Can you please attach a, small, self contained example (source+configuration file+resulting output in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by exyi** [comment]
Ok, I have created a trivial repro: [doxygen-repro-link-to-property.zip](https://github.com/doxygen/doxygen/files/3968343/doxygen-repro-link-to-property.zip)

    public class A { }
    /// This should link to the property A <see cref="A" />
    public class Class1
    {
        public string A { get; }
    }


The output looks like:

![image](https://user-images.githubusercontent.com/7894687/70912488-f5df4280-200b-11ea-8ee4-6155399d8f3f.png)

The link should go to property `A`, since that is how C# compiler links it.

**Comment by albert-github** [comment]
Unfortunately the doxygen configuration file (Doxyfile) is missing, though by mean of a default Doxyfile with just setting EXTRACT_ALL=YES I can reproduce the issue.

It looks like doxygen cannot distinguish between the class `A` and the property `A`, but in fact the property name is not `A` but `Class1.A`, so when specifying `/// This should link to the property Class1.A <see cref="Class1.A" />` the result is OK.

I'm not sure this is a bug.

**Comment by exyi** [comment]
> It looks like doxygen cannot distinguish between the class A and the property A, but in fact the property name is not A but Class1.A, so when specifying /// This should link to the property Class1.A <see cref="Class1.A" /> the result is OK.

> I'm not sure this is a bug.

If Doxygen is supposed to use the standard C# xml-based comments, then it should IMHO behave as similar to the standard C# tools.

**Comment by albert-github** [comment]
The problem with the C# xml-based comments is that they are quite vague in the definitions in the ECMA-334-2017 standard.
For the `<see>`:
> Syntax:
>     `<see cref="member"/>`
> where
>
>  member
>    The name of a member. The documentation generator checks that the given code element exists
and changes member to the element name in the generated documentation file.

the question here what is defined as a `member` is it something in the next class or is it the class as defined above (which is a member of the namespace as shown in your example).

One definition I found in said standard (475
Annex D. Documentation comments on the top of the page 497) is:
> C# provides a mechanism for programmers to document their code using a special comment syntax that contains XML text. In source code files, comments having a certain form can be used to direct a tool to produce XML from those comments and the source code elements, which they precede. Comments using such syntax are called documentation comments. They must immediately precede a user-defined type (such as a class, delegate, or interface) or a member (such as a field, event, property, or method).

this indicates that the `cref="A"` should point to the property  `A` in the class `Class1`.

(as a small side note: standard C# tools can also contain bugs / omissions as a tool is just an interpretation of a standard).

**Comment by exyi** [comment]
Ok, I get the point that the standard is not very specific about that :/ The comment is certainly semantically "inside" the referenced type, but that does not have to mean that the member name can be relative to "current scope". It must only respect the using statements.

They have some examples there that confirm that is behavior is what is intended: https://github.com/dotnet/csharplang/blob/master/spec/documentation-comments.md#see

Maybe we could ask the C# team to add something about it to the standard? They maintain it on github: https://github.com/dotnet/csharplang

**Comment by albert-github** [comment]
Looks like after numerous years a new version of the C# Language Definition (Standard ECMA-334 6th Edition i.e. https://www.ecma-international.org/wp-content/uploads/ECMA-334_6th_edition_june_2022.pdf) has been released.
